pkg/authenticator/authenticator.go

```diff
@@ -236,8 +247,7 @@ func (auth *Authenticator) ParseAuthenticationResponse(response []byte) error {
 		content = response
 	}
 
-	// InfoLogger.Printf("Writing token %v to shared volume ...", content)
-	err = ioutil.WriteFile(auth.Config.TokenFilePath, content, 0644)
+	err = auth.AccessTokenHandler.Write(content)
```

Comment:
I know this is out of scope of this PR. 
The method in which this call takes place has a misleading name (ParseAuthenticationResponse). It's definitely not merely parsing the authentication response.

Reply:
it's very risky to change this now as it's used in external projects

Reply:
That's fair and actually raises an important consideration. External projects using this client really ought to consume an interface so that we can work on the internals free from introducing breaking changes.